llvm/include/llvm/CodeGen/SDPatternMatch.h

```diff
@@ -1299,6 +1299,32 @@ inline BinaryOpc_match<ValTy, AllOnes_match, true> m_Not(const ValTy &V) {
   return m_Xor(V, m_AllOnes());
 }
 
+struct SpecificNeg_match {
+  SDValue V;
+
+  explicit SpecificNeg_match(SDValue V) : V(V) {}
+
+  template <typename MatchContext>
+  bool match(const MatchContext &Ctx, SDValue N) {
+    if (sd_context_match(N, Ctx, m_Neg(m_Specific(V))))
+      return true;
+
```

Comment:
```suggestion
    if (N.getOpcode() == ISD::BUILD_VECTOR && V.getOpcode() == ISD::BUILD_VECTOR)
      return all_of_zip(N->op_values(), V->op_values(),
                        [&](SDValue V1, SDValue V2) { return SpecificNeg_match(V1).match(Ctx, V2); });
```

It is quite useful for my patch to have "specific neg of a bv of consts", and these lines should solve it